Source/DafnyCore/AST/Attributes.cs

```diff
@@ -15,7 +15,32 @@ public static bool IsExplicitAxiom(this IAttributeBearingDeclaration me) =>
     Attributes.Contains(me.Attributes, "axiom");
 }
 
-public class Attributes : TokenNode {
+// Syntax of a formal of a built-in @-attribute
+// To create one, prefer using the chaining BuiltInAtAttributeSyntax.WithArg()
+public record BuiltInAtAttributeArgSyntax(
```

Comment:
This is perfect for now, but I'm curious about the future as well. If we add support for user defined @ attributes in the future, should those behave exactly the same as predefined attributes? In that case, I would expect this record to be replaced by usages of the existing Datatype AST type.

Reply:
You mean, that instead of these records, that we would have a tree with a datatype declaration and datatype constructors direclty?
I think that's indeed the next step before one day being able to refactor them to another file that might be parsed in a prelude. That way, documentation, jumping to definition and other IDE interactions will be easier.
The builder pattern I wrote can be used for that as well.
For now, I like the idea of not touching the user's program yet by injecting definitions.

Reply:
> You mean, that instead of these records, that we would have a tree with a datatype declaration and datatype constructors direclty?

I mean that we'd define the attributes in a .dfy file and they'd get parsed into `DatatypeDecl` and `DatatypeCtor` values.